src/notifications.ts

```diff
@@ -0,0 +1,22 @@
+const Main = imports.ui.main;
+
+import {HistoryEntry} from './history.js';
+
+/**
+ * A convenience class for presenting notifications to the user.
+ */
+class Notification {
+    /**
+     * Show a notification for the newly set wallpapers.
+     *
+     * @param {HistoryEntry[]} historyEntries The history elements representing the new wallpapers
+     */
+    static newWallpaper(historyEntries: HistoryEntry[]): void {
+        const infoString = `Source: ${historyEntries.map(h => `${h.source.source}`).join(', ')}`;
```

Comment:
`h.source.source` can be null which would lead to an awkward looking `Source: , , ` infoString when the map() result is `[null, null, null]`.

Also: Multiple images can have the same source which currently results in duplicate entries in the infoString.

Reply:
That's true. For one, I think we should make sure that field cannot be NULL, but I am also not very satisfied with the notification content, to be honest. It was just the first somewhat useful thing I came up with.

Reply:
So If you have a better suggestion for the notification content then let me know. I can also move that feature out of this PR for now until we have a better Idea.

Reply:
> I think we should make sure that field cannot be NULL

Right now it can be null because pre v3 versions did not contain this key and we might encounter those entries on loading. We could filter old entries easily tho:

https://github.com/ifl0w/RandomWallpaperGnome3/blob/633546ac52836a6faa32a0f7c122594feeb155db/src/history.ts#L278-L297

Might play into #156 

---

> I am also not very satisfied with the notification content

I'm actually fine with the source name. That was the only thing I often checked after changing: Where does this image originate from.

Reply:
Good point! I call it "Unknown Source" in the notification for now.